bin/docserv

```diff
@@ -377,17 +385,44 @@ class DocservState:
                     return deliverable
         return None
 
-    def finished_building_doc(self, doc):
-        self.building_docs_lock.acquire()
-        self.building_docs.pop(doc.__dict__()['id'],None)
-        self.building_docs_lock.release()
+    def get_scheduled_build_instruction(self):
+        # Get a build_instruction for building
+        try:
+            build_instruction = self.scheduled_docs.get(False)
+        except queue.Empty:
+            return None
+        else:
+            return build_instruction
+
+    def store_deliverable_ids(self, build_instruction_id, deliverables):
+        self.in_queues_lock.acquire()
+        self.in_queues[build_instruction_id]['building_deliverables'] = deliverables
+        self.in_queues_lock.release()
```

Comment:
The `.release()` method can be easily forgotten; I'd recommed to use a context manager here as written above.